idea/src/org/jetbrains/kotlin/idea/intentions/RemoveArgumentNameIntention.kt

```diff
@@ -30,7 +32,9 @@ class RemoveArgumentNameIntention : SelfTargetingRangeIntention<KtValueArgument>
 
         val argumentList = element.parent as? KtValueArgumentList ?: return null
         val arguments = argumentList.arguments
-        if (arguments.asSequence().takeWhile { it != element }.any { it.isNamed() }) return null
+        if (!element.languageVersionSettings.supportsFeature(LanguageFeature.MixedNamedArgumentsInTheirOwnPosition)
+            && arguments.asSequence().takeWhile { it != element }.any { it.isNamed() }
+        ) return null
 
         val callExpr = argumentList.parent as? KtCallElement ?: return null
         val argumentMatch = callExpr.resolveToArgumentMatch(element) ?: return null
```

Comment:
You are using your own `resolveToArgumentMatch` function here

Please, replace it with `org.jetbrains.kotlin.resolve.calls.callUtil.getParameterForArgument`, it is designed specifically for this case